Add docs-build session (#498)
cjolowicz/cookiecutter-hypermodern-python: docs/guide.rst

   :ref:`coverage <The coverage session>`     Report coverage with Coverage.py_     ``3.8``
   :ref:`docs <The docs session>`             Build and serve Sphinx_ documentation ``3.8``
   :ref:`docs-build <The docs-build session>` Build Sphinx_ documentation           ``3.8``                ✓
   :ref:`mypy <The mypy session>`             Type-check with mypy_                 ``3.6`` … ``3.8``      ✓
   :ref:`pre-commit <The pre-commit session>` Lint with pre-commit_                 ``3.8``                ✓
   :ref:`safety <The safety session>`         Scan dependencies with Safety_        ``3.8``                ✓
   :ref:`tests <The tests session>`           Run tests with pytest_                ``3.6`` … ``3.8``      ✓
   :ref:`typeguard <The typeguard session>`   Type-check with Typeguard_            ``3.6`` … ``3.8``      ✓
   :ref:`xdoctest <The xdoctest session>`     Run examples with xdoctest_           ``3.6`` … ``3.8``
   ========================================== ===================================== ================== =========

.. _The docs session:

The docs session
----------------

> Build and serve the documentation in `interactive mode`__ using the Nox session ``docs``:

Reviewer: Keep the original simple version:

```suggestion
Build the documentation using the Nox session ``docs``:
```

Keeping it short makes it clear to users that this is what they should be using. The link about Nox's interactive mode does not apply anoymore. The next paragraph explains about the HTTP server, so we can omit that here.

.. code:: console

   $ nox --session=docs

The docs session runs the command ``sphinx-autobuild`` to generate the HTML documentation from the Sphinx directory.

> 

Reviewer: Combine into a single paragraph:

```suggestion
```

> This tool has several advantages when you are editing the documentation files:

Reviewer: ```suggestion
This tool has several advantages over ``sphinx-build`` when you are editing the documentation files:
```

> __ https://nox.thea.codes/en/stable/usage.html#forcing-non-interactive-behavior
> 

Reviewer: Remove obsolete reference.

```suggestion
__ https://nox.thea.codes/en/stable/usage.html#forcing-non-interactive-behavior

```

This Nox session always runs with the current major release of Python.

.. _The docs-build session:

The docs-build session
----------------------

> Build the documentation using the Nox session ``docs-build``:
> 
> .. code:: console
> 
>    $ nox --session=docs-build
> 
> The docs session runs the command ``sphinx-build``
> to generate the HTML documentation from the Sphinx directory.
> 

Reviewer: ```suggestion
The ``docs-build`` session runs the command ``sphinx-build``
to generate the HTML documentation from the Sphinx directory.

This session is meant to be run as a part of automated checks.
Use the interactive ``docs`` session instead while you're editing the documentation.

```

> Use the ``--`` separator to pass additional options to either tool.

Reviewer: ```suggestion
Use the ``--`` separator to pass additional options.
```

Please also apply this change (remove "to either tool") to the section about the `docs` session.

Reviewer: **Update:** Actually, please remove this line from the section about `docs-build`. But do include it in the section about `docs`, without the text `"to either tool"`.

(Sorry, been a bit tricky to do this with inline suggestions...)

Author: No problem, I don't even know how to do this inline reviews yet. Everything coming out in one minute.

> For example, to treat warnings as errors and run in nit-picky mode:
> 
> .. code:: console
> 
>    $ nox --session=docs-build -- -W -n docs docs/_build

Reviewer: We can omit the example here, since it is already included in the section about the `docs` session.

```suggestion
```
